infra/swapi.go

```diff
@@ -0,0 +1,80 @@
+package infra
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strconv"
+
+	"github.com/SergioSanchezWizeline/academy-go-q12021/domain/model"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type SwapiPeople struct {
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
+
+type SwapiResult struct {
+	Count   int           `json:"count"`
+	Next    string        `json:"next"`
+	Results []SwapiPeople `json:"results"`
+}
+
+func LoadSwapiCharacters() ([]*model.Character, error) {
+
+	jobs := make(chan string, 10)
+	people := make(chan SwapiPeople, 100)
+
+	fmt.Println("Fetch start")
+
+	jobs <- "https://swapi.dev/api/people/"
+
+	client := resty.New()
+
+	go worker(client, jobs, people)
+	go worker(client, jobs, people)
```

Comment:
Why send two goroutines initially?

Reply:
My idea here was to have two workers to speed up the fetch process, each worker add jobs, so in theory as soon the first worker add a new job , the second will start without waiting for the first one to finish.